src/transformStateWithClones.js

```diff
@@ -7,7 +7,24 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateArr = [];
+  const newState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(newState, action.extraData);
+    } else if (action.type === 'removeProperties') {
+      action.keysToRemove.forEach(i => delete newState[i]);
+    } else {
+      Object.keys(newState).forEach(key => delete newState[key]);
+    }
```

Comment:
you'd rather use switch-case
```
  for (const action of actions) {
    switch(action.type) {
      case 'addProperties':
        ...
      case 'removeProperties':
        ...
      case 'clear': // here we indicate case explicitly
        ...
      default: // everything else – doesn't matter
        break;
    }
  }
```
why? simpler to use + in case what you've written – that's not obvious what we do there